utilities/tools/Test-NamePrefixAvailability.ps1

```diff
@@ -56,123 +63,136 @@ function Test-NamePrefixAvailability {
-
-        try {
-
-            # Extract Parameter Names
-            # -----------------------
-            $storageAccountFiles = $parameterFiles | Where-Object { $_ -match 'Microsoft.Storage/storageAccounts' }
-            $storageAccountNames = $storageAccountFiles | ForEach-Object { (ConvertFrom-Json (Get-Content $_ -Raw)).parameters.name.value } | Where-Object { $null -ne $_ }
-
-            $keyVaultFiles = $parameterFiles | Where-Object { $_ -match 'Microsoft.KeyVault/vaults' }
-            $keyVaultNames = $keyVaultFiles | ForEach-Object { (ConvertFrom-Json (Get-Content $_ -Raw)).parameters.name.value } | Where-Object { $null -ne $_ }
 
-            $acrFiles = $parameterFiles | Where-Object { $_ -match 'Microsoft.ContainerRegistry/registries' }
-            $acrNames = $acrFiles | ForEach-Object { (ConvertFrom-Json (Get-Content $_ -Raw)).parameters.name.value } | Where-Object { $null -ne $_ }
+        $storageAccountNames = @()
+        $containerRegistryNames = @()
+        $keyVaultNames = @()
 
-            $subscriptionId = (Get-AzContext).Subscription.Id
+        $parameterFiles = (Get-ChildItem -Path $repoRoot -Recurse -Filter 'deploy.test.bicep').FullName | ForEach-Object { $_.Replace('\', '/') }
 
-            # Storage
-            Write-Host "`nCheck Storage Accounts" -ForegroundColor 'Cyan'
-            Write-Host '======================' -ForegroundColor 'Cyan'
-            foreach ($storageAccountName in $storageAccountNames) {
-                $path = '/subscriptions/{0}/providers/Microsoft.Storage/checkNameAvailability?api-version=2021-04-01' -f $subscriptionId
-                $requestInputObject = @{
-                    Method  = 'POST'
-                    Path    = $path
-                    Payload = @{
-                        name = $storageAccountName
-                        type = 'Microsoft.Storage/storageAccounts'
-                    } | ConvertTo-Json
+        foreach ($relevantResourceType in $relevantResourceTypes) {
+            switch ($relevantResourceType) {
+                'Microsoft.Storage/storageAccounts' {
+                    $filter = 'storageAccountName:'
                 }
-                $response = ((Invoke-AzRestMethod @requestInputObject).Content | ConvertFrom-Json)
-                if ($response.nameAvailable) { Write-Host "=> The storage account name [$storageAccountName] is currently available" -ForegroundColor 'Green' }
-                else {
-                    Write-Warning "=> Warning: The storage account name [$storageAccountName] is not available. Please try a different prefix."
-                    $prefixAvailable = $false
+                'Microsoft.ContainerRegistry/registries' {
+                    $filter = 'registryName:'
                 }
+                'Microsoft.KeyVault/vaults' {
+                    $filter = 'keyVaultName:'
+                }
+                Default { Write-Error 'I dont like you.' }
```

Comment:
❤️

Reply:
I've updated this one to a more boring but better understandable error message.